spec/controllers/catalog_controller_spec.rb

```diff
@@ -152,15 +155,146 @@
+      it 'leaves the cart alone if items is a Hash' do
+        session[:cart] = FactoryGirl.build(:cart_with_items)
+        expect { get :index }.to_not change { session[:cart].items }
+      end
+    end
+
+    describe 'app_setup_check' do
+      context 'user and appconfig in the db' do
+        before(:each) do
+          get :index
+        end
+        it { is_expected.to respond_with(:success) }
+        it { is_expected.not_to set_flash }
+      end
+      context 'no app_config' do
+        before(:each) do
+          AppConfig.delete_all
+          get :index
+        end
+        it { is_expected.to set_flash }
+        it { is_expected.to render_template('application_setup/index') }
+      end
+      context 'no user in the db' do
+        before(:each) do
+          sign_out @user
+          User.delete_all
+          get :index
+        end
+        it { is_expected.to set_flash }
+        it { is_expected.to render_template('application_setup/index') }
+      end
+    end
+
+    describe 'seen_app_configs' do
+      before(:each) do
+        @admin = FactoryGirl.create(:admin)
+        sign_in @admin
+      end
+      context 'app configs have not been viewed' do
+        before(:each) do
+          AppConfig.delete_all
+          @app_config = FactoryGirl.create(:app_config, viewed: false)
+          get :index
+        end
+        it { is_expected.to set_flash }
+        it { is_expected.to respond_with(:redirect) }
+        it { is_expected.to redirect_to(edit_app_configs_path) }
+      end
+      context 'app configs have been viewed' do
+        before(:each) do
+          get :index
+        end
+        it { is_expected.not_to set_flash }
+        it { is_expected.to respond_with(:success) }
+        it { is_expected.not_to redirect_to(edit_app_configs_path) }
+      end
+    end
+
+    describe 'load_configs' do
+      it 'should set @app_configs to the first AppConfig' do
+        get :index
+        expect(assigns(:app_configs)).to eq(AppConfig.first)
+      end
+    end
+
+    describe 'cart' do
+      it 'makes a new cart record for session[:cart] if !cart' do
+        get :index
+        expect(session[:cart].is_a?(Cart)).to be_truthy
+      end
+      it 'returns session[:cart] if cart.reserver_id' do
+        session[:cart] = Cart.new
+        session[:cart].reserver_id = @user.id
+        get :index
+        expect(session[:cart].reserver_id).to eq(@user.id)
+      end
+      it 'sets the session[:cart].reserver_id to current_user.id if '\
+        '!cart.reserver_id && current_user' do
+        session[:cart] = Cart.new
+        get :index
+        expect(session[:cart].reserver_id).to eq(@user.id)
+      end
+      it 'returns session[:cart] without a reserver_id if !cart.reserver_id '\
+        '&& !current_user' do
+        sign_out @user
+        session[:cart] = Cart.new
+        get :index
+        expect(session[:cart].reserver_id).to be_nil
+      end
+    end
+
+    describe 'set_view_mode' do
```

Comment:
I know that this was copied over, but do we even need this? Alternatively, should we be writing tests?